src/Microsoft.PowerShell.ConsoleGuiTools/ConsoleGui.cs

```diff
@@ -267,6 +272,8 @@ private void FilterData(string filter)
                 if (match)
                 {
                     items.Add(GetPaddedString(valueList));
```

Comment:
If `items` was an `ArrayList`, the `Add` would return the index that you would put right into:
```
_indexMap.Add(index, i);
```

(in other words, no `index` variable)

Reply:
If you don't want to do that, can you at least rename `index` to `newIndex` so we know it's the new index of the list?

Reply:
I'll go with the latter since ArrayList is not recommended